src/test/java/com/library/utils/IsSameLikeEntity.java

```diff
@@ -0,0 +1,28 @@
+package com.library.utils;
+
+import com.library.model.entity.Book;
+import com.library.model.entity.ParentEntity;
+import org.mockito.ArgumentMatcher;
+
+public class IsSameLikeEntity<T extends ParentEntity> implements ArgumentMatcher<T> {
+
+    private final T t;
+    public IsSameLikeEntity(T t) {
+        this.t = t;
+    }
+
+    @Override
+    public boolean matches(T argument) {
+        String className = argument.getClass().getSimpleName();
+        if ("Book".equals(className)) {
```

Comment:
Wrong realization.

What SOLID principle is not observed?
What will be if it is needed to check similar things for another entity?